osquery/tables/applications/darwin/browser_plugins.cpp

```diff
@@ -152,45 +147,59 @@ inline void genSafariExtension(const std::string& uid,
     return;
   }
 
-  // Use open_file, instead of the preferred open_filename for OS X 10.9.
-  archive_read_support_format_xar(ext);
-  if (archive_read_open_filename(ext, path.c_str(), 10240) != ARCHIVE_OK) {
-    archive_read_finish(ext);
+  xar_t xar = xar_open(path.c_str(), READ);
+  if (xar == nullptr) {
```

Comment:
Actually, the current implementation is OK. There may be non-extension files within the directory. They should not be reported as black entries.